bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -0,0 +1,738 @@
+                }
+                properties.put(PROPERTY_RESOURCE_ARCHETYPE, metaData.getArchetype().toString());
+
+                // set product data properties
+                ProductData productData = device.getProductData();
+                if (Objects.nonNull(productData)) {
+                    // set generic thing properties
+                    properties.put(Thing.PROPERTY_MODEL_ID, productData.getModelId());
+                    properties.put(Thing.PROPERTY_VENDOR, productData.getManufacturerName());
+                    properties.put(Thing.PROPERTY_FIRMWARE_VERSION, productData.getSoftwareVersion());
+                    String hardwarePlatformType = productData.getHardwarePlatformType();
+                    if (Objects.nonNull(hardwarePlatformType)) {
+                        properties.put(Thing.PROPERTY_HARDWARE_VERSION, hardwarePlatformType);
+                    }
+
+                    // set hue specific properties
+                    properties.put(PROPERTY_PRODUCT_NAME, productData.getProductName());
+                    properties.put(PROPERTY_PRODUCT_ARCHETYPE, productData.getProductArchetype().toString());
+                    properties.put(PROPERTY_PRODUCT_CERTIFIED, productData.getCertified().toString());
+                }
+                break; // we only needed the BRIDGE_V2 resource
+            }
+        }
+        thing.setProperties(properties);
+    }
+
+    /**
+     * Update the thing's own state. Called sporadically in case any SSE events may have been lost.
+     */
+    private void updateSelf() {
+        logger.debug("updateSelf()");
+        try {
+            checkAssetsLoaded();
+            updateProperties();
+            updateStatus(ThingStatus.UNKNOWN);
+            getClip2Bridge().open();
+        } catch (ApiException e) {
+            logger.debug("updateSelf() {}", e.getMessage(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/offline.communication-error");
+            onConnectionOffline();
+        } catch (AssetNotLoadedException e) {
+            logger.debug("updateSelf() {}", e.getMessage(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.api2.conf-error-assets-not-loaded");
+        } catch (HttpUnauthorizedException e) {
+            logger.debug("updateSelf() {}", e.getMessage(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.api2.conf-error-access_denied");
+        }
+    }
+
+    /**
+     * Check if a PROPERTY_LEGACY_THING_UID value was set by the discovery process, and if so, clone the legacy thing's
+     * settings into this thing.
+     */
+    private void updateThingFromLegacy() {
+        if (isInitialized()) {
+            logger.warn("updateThingFromLegacy() was called after handler was initialized.");
```

Comment:
I suppose it would be a bug if the user would see it. So far I never saw it. But it is probably worth a warning if it would arise. I will add the text 'please contact maintainer' ..

Reply:
I think this was forgotten, or perhaps you changed your mind?